server/sonar-web/src/main/js/components/SelectList/SelectList.tsx

```diff
@@ -108,12 +112,13 @@ export default class SelectList extends React.PureComponent<Props, State> {
           />
         </div>
         <SelectListListContainer
+          disabledElements={this.props.disabledElements || []}
           elements={this.props.elements}
-          filter={filter}
+          filter={this.state.query === '' ? filter : Filter.All}
```

Comment:
I'd extract the expression `this.state.query === '' ? filter : Filter.All` to a new method:
```javascript
getFilter = () => {
  return this.state.query === '' ? this.state.filter : Filter.All
}
```

and use it here and in `handleQueryChange `